std/range/package.d

```diff
@@ -1849,19 +1849,30 @@ if (isRandomAccessRange!(Unqual!R) && hasLength!(Unqual!R) && hasSlicing!(Unqual
 Lazily takes only up to `n` elements of a range. This is
 particularly useful when using with infinite ranges.
 
-Unlike $(LREF takeExactly), `take` does not require that there
-are `n` or more elements in `r`. As a consequence, length
-information is not applied to the result unless `r` also has
+Unlike `takeExactly`, `take` does not require that there
+are `n` or more elements in `input`. As a consequence, length
+information is not applied to the result unless `input` also has
 length information.
 
 Params:
-    r = an input range to iterate over up to `n` times
+    input = an input range to iterate over up to `n` times
     n = the number of elements to take
 
 Returns:
     At minimum, an input range. If the range offers random access
     and `length`, `take` offers them as well.
  */
+Take!R take(R)(R input, size_t n)
+if (isInputRange!(Unqual!R) && !isInfinite!(Unqual!R) && hasSlicing!(Unqual!R) &&
+    !is(R T == Take!T))
+{
+    // @@@BUG@@@
+    //return input[0 .. min(n, $)];
```

Comment:
> I thought that $ and length are equivalent.

Unfortunately they are not - `opDollar` needs to be defined separately :/
The rationale is that thus one can have special overloads that can be faster.
However, this makes using `$` very, very error-prone - even in Phobos:

https://issues.dlang.org/show_bug.cgi?id=16073

>  IMHO we shouldn't cater to them.

Yes also it would be nice if `opDollar` would automatically alias to `length` if not defined and there is `length` defined.

> I am starting to think that this bug comment is a residue

Well kind of. It's still true that `opDollar` simply might not be defined if `hasSlicing` is `true`, but there's no point of having this comment there
-> leave as it is currently and just remove the bug comment ;-)